### ISSUE TITLE
starting ephemeral containers

### ISSUE THREAD

**Issue by akshaykarle** [lxc/ruby-lxc#33]
Anyway to replicate the cli `lxc-start-ephemeral` using the gem?

**Comment by andrenth** [comment]
Well, lxc-start-ephemeral is a Python script, and the Ruby LXC bindings should be equivalent in functionality to Python's. I don't see why it couldn't be translated to Ruby.

**Comment by ranjib** [comment]
Yup, currently ruby-lxc only provides Ruby API for liblxc. Akshay if you
want to take stab at it that will be awesome, else I'll try implementing it
during this weekend. We can tag this issue as enhancement.

**Comment by akshaykarle** [comment]
Yes, I had a look at the implementation of lxc-start-ephemeral and I'm already trying to replicate the same in ruby-lxc. Will keep you all updated about the progress and if I need any help. @ranjib I'll get in touch with you if I need anything. Thanks!

**Comment by akshaykarle** [comment]
I [forked](https://github.com/akshaykarle/ruby-lxc/tree/start-ephemeral) and added my changes for starting ephemeral containers. I have a basic functionality working with overlayfs and before adding support for other options (tmpfs, aufs, commands, etc.) I wanted to write tests and clean up the code as right now it is simply a port of lxc-start-ephemeral script and is quite ugly.

But I'm having trouble running the tests which are failing to run when trying to require 'lxc/lxc'. You can check out this https://gist.github.com/akshaykarle/6e140b345f0a42b2e358 for more details on the failure. I also had a look at the travis yml to see if it had any manual setup but it doesn't even seem to run the tests. I've already installed lxc-dev and ruby-dev on the machine. Can someone help me getting the tests running? Thanks!
